src/cluster_legacy.c

```diff
@@ -6543,7 +6556,30 @@ unsigned int delKeysInSlot(unsigned int hashslot, int lazy, bool propagate_del,
      * state so that we don't assert in propagateNow(). */
     server.server_del_keys_in_slot = 1;
     unsigned int j = 0;
+    robj *argv[4];
     int before_execution_nesting = server.execution_nesting;
+    enterExecutionUnit(1, 0);
```

Comment:
Do we need two execution units here? It seems like just one would be enough to cover both cases.